src/Eurofurence.App.Server.Services/Events/EventFavoritesService.cs

```
@@ -0,0 +1,67 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Eurofurence.App.Domain.Model.Events;
+using Eurofurence.App.Domain.Model.PushNotifications;
+using Eurofurence.App.Infrastructure.EntityFramework;
+using Eurofurence.App.Server.Services.Abstractions;
+using Eurofurence.App.Server.Services.Abstractions.Events;
+using Eurofurence.App.Server.Services.Abstractions.Security;
+using Microsoft.EntityFrameworkCore;
+
+namespace Eurofurence.App.Server.Services.Events
+{
+    public class EventFavoritesService(
+        AppDbContext appDbContext,
+        IStorageServiceFactory storageServiceFactory,
+        bool useSoftDelete = true)
+        : IEventFavoritesService
+    {
+        public async Task AddEventToFavoritesIfNotExist(ClaimsPrincipal user, EventRecord eventRecords)
+        {
+            if (!appDbContext.Users.Any(x =>
```

The outer if statement could be removed, instead we could use FirstOrDefault and throw an error if the user is not found

---

I think the `user` object is never null since we use this method in an Authorized endpoint.
But a sanity check can't hurt.